Adding documentation on Adapter. (#104)
refractionPOINT/lce_doc: docs/lc_adapter.md

## Availability

* Docker: https://hub.docker.com/r/refractionpoint/lc-adapter
* Various Executables:

>   * [Windows x64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_windows_64_1.2.2.exe)

Reviewer: ```suggestion
  * [Windows x64](https://app.limacharlie.io/get/adapter/win64)
```

Author: Thanks! Don't know what I was thinking. :)

>   * [macOS x64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_macos_64_1.2.2)

Reviewer: ```suggestion
  * [macOS x64](https://app.limacharlie.io/get/adapter/mac64)
```

>   * [macOS arm64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_macos_arm64_1.2.2)

Reviewer: ```suggestion
  * [macOS arm64](https://app.limacharlie.io/get/adapter/macarm64)
```

>   * [Linux x64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_linux_64_1.2.2)

Reviewer: ```suggestion
  * [Linux x64](https://app.limacharlie.io/get/adapter/linux64)
```

  * [Let us know if you need a specific platform.](https://www.limacharlie.io/contact)

## Concept
The LimaCharlie Adapter is a piece of software that forwards any logs or telemetry of your choice to LimaCharlie in real-time.
It can be run on any platform and in any location (on premises for example).

The Adapter may itself get the logs/telemetry any number of locations and using many different methods like:

* Syslog
* AWS S3

> * Google Cloud Pubsub

Reviewer: ```suggestion
* Google Cloud Pub/Sub
```

* `client_options.identity.oid=....`: the Organization ID from LimaCharlie the installation key above belongs to.
* `client_options.platform=text`: this indicates the type of data that will be received from this adapter. In this case it's syslog, so `text` lines.
* `client_options.mapping.parsing_re=....`: this is the parsing expression describing how to interpret the text lines and how to convert them to JSON.
* `client_options.sensor_seed_key=....`: this is the value that identifies this instance of the Adapter. Record it to re-use the Sensor ID generated for this Adapter later if you have to re-install the Adapter.
* `client_options.mapping.rename_only=true`: only rename the field in mapping below, so keep the other original fields.
* `client_options.mapping.mapping[0].src_field=....`: the source field of the first mapping record.
* `client_options.mapping.mapping[0].dst_field=....`: the destination field of the first mapping record.

To test it, assuming we're on the same Debian box as the container, pipe the syslog to the container:
```
journalctl -f -q | netcat 127.0.0.1 4444
```

### CarbonBlack from S3

> This example shows connecting CarbonBlack sensors from data exported by the CarbonBlack API to an S3 bucket.

Reviewer: ```suggestion
This example shows connecting Carbon Black sensors from data exported by the Carbon Black API to an S3 bucket.
```

It uses the CLI Adapter (instead of the Docker container).

```
./lc_adapter s3 client_options.identity.installation_key=e9a3bcdf-efa2-47ae-b6df-579a02f3a54d client_options.identity.oid=8cbe27f4-bfa1-4afb-ba19-138cd51389cd client_options.platform=carbon_black client_options.sensor_seed_key=tests3 bucket_name=lc-cb-test access_key=YYYYYYYYYY secret_key=XXXXXXXX  "prefix=events/org_key=NKZAAAEM/"
```

Here's a breakdwn of the above example:

* `lc_adapter`: simply the CLI Adapter.
* `s3`: the data will be collected from an AWS S3 bucket.
* `client_options.identity.installation_key=....`: the installation key value from LimaCharlie.
* `client_options.identity.oid=....`: the Organization ID from LimaCharlie the installation key above belongs to.

> * `client_options.platform=carbon_black`: this indicates the data received will be CarbonBlack events from their API.

Reviewer: ```suggestion
* `client_options.platform=carbon_black`: this indicates the data received will be Carbon Black events from their API.
```

> * `client_options.sensor_seed_key=....`: this is the value that identifies this instance of the Adapter. Record it to re-use the Sensor IDs generated for the CarbonBlack sensors from this Adapter later if you have to re-install the Adapter.

Reviewer: ```suggestion
* `client_options.sensor_seed_key=....`: this is the value that identifies this instance of the Adapter. Record it to re-use the Sensor IDs generated for the Carbon Black sensors from this Adapter later if you have to re-install the Adapter.
```

* `bucket_name:....`: the name of the S3 bucket holding the data.
* `secret_key:....`: the API key for AWS that has access to this bucket.

> * `prefix=....`: the file/directory name prefix that holds the CarbonBlack data within the bucket.

Reviewer: ```suggestion
* `prefix=....`: the file/directory name prefix that holds the Carbon Black data within the bucket.
```
